### PR TITLE
Pass multiprocessing parameters through the shell

### PR DESCRIPTION
**Pull request by dcmaddix**
### Summary
- Exposed the multiprocessing parameters `num_workers`, `num_prefetch`, `shuffle_buffer_length` as `hyperparameter` keys through the shell
- Updated `run_train` method to read these keys in from the hyper-parameters and set their values
- For `num_workers` we set to be the minimum of inputed value if set and if not input default 4 and the sort of the number of cpus
- Added logging to log these parameter values in`parallelized_loader.py`
- Added logging to whether the dataset was cached or not and its type, i.e. `ListDataset` and `FileDataset`, respectively

### Testing
- Add `MQCNNEstimator` as `forecaster_type` to `test_train_shell` to extend the test for trainable algorithms unlike the current `MeanPredictor`
- Passed the three multiprocessing parameters as hyper-parameters and used the smoke test with the `MQCNNEstimator`

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by dcmaddix** [comment]
There is a timeout test error of 30s for the additional test I added with the `MQCNNEstimator`.  It takes longer than that to run.  Is there a way to increase the test timeout>
